### PR TITLE
Fix for wp-missed-scheduled plugin

### PR DESCRIPTION
**Pull request by enejb**
Fixes the compatibility with the wp-missed-scheduled plugin. 

The plugin published scheduled posts on certain hosts where cron is not working as expected.

To test:
Disable cron add to wp-config.php
define( 'DISABLE_WP_CRON', true );

Add the plugin https://github.com/sLaNGjI/wp-missed-schedule

Schedule a new post.
It should to email the subscribers. ( subscriptions should be on)

The subscribers should get an email.After the post is published.

### CODE REVIEW

**Comment by enejb** [comment]
Note to self: this PR needs to make sure that the whitelist filters added by Akismet, VaultPress, etc Other plugins still work as expected. :/

File: sync/class.jetpack-sync-actions.php [hunk 1]
```diff
@@ -341,11 +345,7 @@ static function init_sync_cron_jobs() {
  * If the site is using a regular cron job, we init at a priority of 90 which gives plugins a chance
  * to add filters before we initialize.
  */
-if ( defined( 'ALTERNATE_WP_CRON' ) && ALTERNATE_WP_CRON ) {
-	add_action( 'init', array( 'Jetpack_Sync_Actions', 'init' ), 9 );
-} else {
-	add_action( 'init', array( 'Jetpack_Sync_Actions', 'init' ), 90 );
-}
+add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'init' ), 90 );
```
Comment (lezama) [inline sync/class.jetpack-sync-actions.php]: Why 90?

Comment (enejb) [inline sync/class.jetpack-sync-actions.php]: no particular reason.

**Comment by jeherve** [comment]
@enejb Will that help when using other plugins, like [Scheduled Post Trigger](https://wordpress.org/plugins/scheduled-post-trigger/)?

**Comment by enejb** [comment]
Yes it should since we would be loading the listener super early. Scheduled Post Trigger should be working with Jetpack 4.4 though at least in my testing.

**Comment by gravityrail** [comment]
LGTM! 👍

**Comment by gravityrail** [comment]
I just had a thought about this: In jetpack-sync-defaults.php we load the available "sync modules" from the get_modules() method, and they are only initialized once. Depending on when we first fetch a module, this may prevent some plugins that wish to add their own modules from doing so... unless they initialize the filter at plugins_loaded 89 :/ Something to think about.

**Comment by gravityrail** [comment]
So basically there's a trade-off here - the earlier we initialize the listener, the more things it will be able to "catch" (if plugins are firing off synced actions during init, which seems BAD) but the harder it will be for anyone else to hook into sync.

**Comment by enejb** [comment]
I think this is ready to merge. Now I couldn't find any other filers or actions that would be effected by the change. Other then the whitelist_options filter.

**Pull request merged**